src/Microsoft.OData.Client/BulkUpdateSaveResult.cs

```diff
@@ -53,15 +86,31 @@ internal BulkUpdateGraph BulkUpdateGraph
             get { return this.bulkUpdateGraph; }
         }
 
-        protected override Stream ResponseStream => throw new NotImplementedException();
+        /// <summary>
+        /// Returns an instance of the link descriptors.
+        /// </summary>
+        internal Dictionary<Descriptor, HashSet<LinkDescriptor>> LinkDescriptors
+        {
+            get { return this.linkDescriptors; }
+        }
+
+        protected override Stream ResponseStream
+        {
+            get { return this.responseStream; }
+        }
 
         /// <summary>
-        /// Syncronous bulk update request.
+        /// Synchronous bulk update request.
         /// </summary>
         /// <typeparam name="T"> The type of the top-level objects to be deep-updated.</typeparam>
         /// <param name="objects"> The top-level objects of the type to be deep updated.</param>
         internal void BulkUpdateRequest<T>(params T[] objects)
         {
+            if (objects == null || objects.Length == 0)
+            {
+                throw Error.Argument(Strings.Util_EmptyArray, "resources");

```

Comment:
You misunderstood a comment I left here sometimes back https://github.com/OData/odata.net/pull/2627#discussion_r1166310189

I meant the "top-level objects" literal that was there before should be moved to the resource file. I didn't mean that you should replace it with the literal "resources"

Reply:
This is a resource that expects a parameter. Are you saying we have that parameter in a resource? What will be the benefit?

Reply:
@ElizabethOkerio When I first left the comment I referred to, here's what was on this line:
```csharp
throw Error.Argument(Strings.Util_EmptyArray, "top-level objects");
```
To me, "top-level objects" was a literal and the proper place to stash it was in the resources file.
But if the idea is to alert about the `objects` parameter being empty, then the following would suffice:
```csharp
throw Error.Argument(Strings.Util_EmptyArray, "objects");
```
or even better
```csharp
throw Error.Argument(Strings.Util_EmptyArray, nameof(objects));
```
Where the second parameter is referring to the parameter that is passed into the method.